### PR TITLE
feat: US130195 - focus fixes and open profile on enter key

### PR DESCRIPTION
**Pull request by annabelsegalld2l**

### CODE REVIEW

File: test/user-profile-card.visual-diff.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ describe('d2l-labs-user-profile-card', () => {
 					{ once: true }
 				);
 			});
+			// force focus on the name to prevent diff flakes
+			elem._focusName = true;
```
Comment (annabelsegalld2l) [inline test/user-profile-card.visual-diff.js:33]: I was having trouble with some flakes where the name was not getting focused on, so I just forced it on here for the visual diffs. It seemed like the flake only happens on the demo page if you open and close super quickly, but it was happening around half the time with the tests.

File: user-profile-card.js [hunk 2]
```diff
@@ -132,18 +133,20 @@ class UserProfileCard extends LocalizeUserProfileCard(RtlMixin(LitElement)) {
...
+					@d2l-dropdown-close="${this._onDropdownClosed}"
+					@d2l-dropdown-focus-enter="${this._onDropdownFocusEnter}">
 						<div class="d2l-labs-profile-card">
 							<d2l-offscreen role="alert">${openAlert}</d2l-offscreen>
-							<div class="d2l-labs-profile-card-image-wrapper"
-									@click="${this._openUserProfile}">
+							<div 
+								class="d2l-labs-profile-card-image-wrapper"
+								@click="${this._openUserProfile}"
+								@keydown="${this._onUserProfileKeypress}">
```
Comment (annabelsegalld2l) [inline user-profile-card.js:149]: Adding a listener for keydown events on the profile image and name link so they can open on enter as well as clicking.

File: user-profile-card.js [hunk 3]
```diff
@@ -212,9 +222,15 @@ class UserProfileCard extends LocalizeUserProfileCard(RtlMixin(LitElement)) {
 	}
 
 	_onDropdownClosed() {
+		this._focusName = false;
 		this._isOpen = false;
 	}
 
+	_onDropdownFocusEnter(e) {
+		// only focus on keyboard opening events
+		this._focusName = e.detail?.applyFocus;
```
Comment (annabelsegalld2l) [inline user-profile-card.js:231]: I just added an applyFocus detail that allows us to know whether or not we need to override focus on the card. Ie. we can focus on the name for keyboard events but not on click or hover events.

Comment (svanherk) [reply user-profile-card.js:231]: Will this cause the same CI problem we were seeing in activities?

Comment (annabelsegalld2l) [reply user-profile-card.js:231]: Yes it will. Good catch ✔️

**Review by svanherk — APPROVED** [review]
Nice!

**Pull request merged**